### PR TITLE
Feat(dashboard): admin locations v2

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Add initial locations components for routes-v2

**remaining work**
Sales channel location management
- this is not supported by the api and will follow in subsequent pr's

### CODE REVIEW

File: packages/admin-next/dashboard/src/v2-routes/locations/location-create/components/create-location-form/create-location-form.tsx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+import { RouteFocusModal } from "../../../../../components/route-modal"
+
+const CreateLocationSchema = zod.object({
+  name: zod.string().min(1),
+  address: zod.object({
+    address_1: zod.string().min(1),
+    address_2: zod.string().optional(),
+    country_code: zod.string().min(2).max(2),
+    city: zod.string().optional(),
+    postal_code: zod.string().optional(),
```
Comment (riqwan) [inline packages/admin-next/dashboard/src/v2-routes/locations/location-create/components/create-location-form/create-location-form.tsx]: q: shouldn't this be required?

**Review by riqwan — APPROVED** [review]
lgtm!

File: packages/admin-next/dashboard/src/v2-routes/locations/location-create/components/create-location-form/create-location-form.tsx (hunk 1 above)
Comment (pKorsholm) [reply packages/admin-next/dashboard/src/v2-routes/locations/location-create/components/create-location-form/create-location-form.tsx]: it's not required in the api or the db, then we should align what we require and what we don't across frontend and backend 😄

File: packages/admin-next/dashboard/src/components/layout/shell/shell.tsx [hunk 2]
```diff
@@ -131,6 +132,9 @@ const UserBadge = () => {
     enabled: isV2Enabled,
   })
 
+  console.warn(v1)
+  console.warn(v2)
+
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/components/layout/shell/shell.tsx]: todo: rm

File: packages/admin-next/dashboard/src/v2-routes/locations/location-detail/location-detail.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import { LocationGeneralSection } from "../../../modules/locations/location-detail/components/location-general-section"
+import { LocationSalesChannelSection } from "../../../modules/locations/location-detail/components/location-sales-channel-section"
+import { useAdminStockLocations } from "medusa-react"
+
+export const LocationDetail = () => {
+  const { id } = useParams()
+  const { stock_locations, isLoading, isError, error } = useAdminStockLocations(
+    {
+      id,
+      expand: "address,sales_channels",
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/v2-routes/locations/location-detail/location-detail.tsx]: todo: shouldn't we use the new API?

File: packages/admin-next/dashboard/src/v2-routes/locations/location-edit/location-edit.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import { useParams } from "react-router-dom"
+import { useTranslation } from "react-i18next"
+
+export const LocationEdit = () => {
+  const { id } = useParams()
+
+  const { stock_locations, isLoading, isError, error } = useAdminStockLocations(
+    {
+      id,
+      expand: "address",
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/v2-routes/locations/location-edit/location-edit.tsx:14]: todo: same for all expand

File: packages/admin-next/dashboard/src/v2-routes/locations/location-edit/location-edit.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  const stock_location = stock_locations?.[0]
+
+  return (
+    <RouteDrawer>
+      <RouteDrawer.Header>
+        <Heading className="capitalize">{t("locations.editLocation")}</Heading>
+      </RouteDrawer.Header>
+      {!isLoading && stock_location && (
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/v2-routes/locations/location-edit/location-edit.tsx]:
```suggestion
      {!isLoading && !!stock_location && (
```

File: packages/admin-next/dashboard/src/v2-routes/locations/location-edit/location-edit.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  const { id } = useParams()
+
+  const { stock_locations, isLoading, isError, error } = useAdminStockLocations(
+    {
+      id,
+      expand: "address",
+    }
+  )
+
+  const { t } = useTranslation()
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/v2-routes/locations/location-edit/location-edit.tsx:18]: suggestion: maybe we should move that after checking the error?

**Review by adrien2p — APPROVED** [review]
LGTM, few comments

File: packages/admin-next/dashboard/src/v2-routes/locations/location-edit/location-edit.tsx (hunk 6 above)
Comment (kasperkristensen) [reply packages/admin-next/dashboard/src/v2-routes/locations/location-edit/location-edit.tsx:18]: hooks should not be used conditionally, they need to be used in a safe and predictable way so they need to go before any conditional check or errors being thrown. So it should stay where it is 😄

File: packages/admin-next/dashboard/src/v2-routes/locations/location-detail/location-detail.tsx (hunk 3 above)
Comment (pKorsholm) [reply packages/admin-next/dashboard/src/v2-routes/locations/location-detail/location-detail.tsx]: the api's for stock locations are not changed, this hits the new api endpoint :) or do you mean using `fields` and not `expand`?

Comment (adrien2p) [reply packages/admin-next/dashboard/src/v2-routes/locations/location-detail/location-detail.tsx]: Yes 😅

**Pull request merged**